tools/roxie/extract-roxie-timings.py

```diff
@@ -0,0 +1,241 @@
+    parser.add_argument("--summaryonly", "-s", help="Only generate a summary", action='store_true')
+    args = parser.parse_args()
+    combineServices = args.all
+    suppressDetails = args.summaryonly
+    reportSummary = not args.nosummary or args.summayonly
+
+    csv.field_size_limit(0x100000)
+    with open(args.filename) as csv_file:
+        csv_reader = csv.reader(csv_file, delimiter=' ')
+        line_count = 0
+        for row in csv_reader:
+            numCols = len(row);
+            rowText = row[numCols-1]
+
+            completeMatch = completePattern.search(rowText)
+            if completeMatch:
+                curRow = dict();
+                mapping = rowText.split();
+
+                serviceName = completeMatch.group(1)
+                idMatch = idPattern.search(mapping[0])
+                if idMatch:
+                    if combineServices:
+                        curRow["_id_"] = serviceName + ':' + idMatch.group(1)
+                    else:
+                        curRow["_id_"] = idMatch.group(1)
+                else:
+                    curRow["_id_"] = mapping[0].replace(",","!");
+
+                elapsedMatch = elapsedPattern.search(rowText)
+                if elapsedMatch:
+                    curRow["elapsed"] =  int(elapsedMatch.group(1))
+
+                #MORE: Unimplemented - allow timestamp filtering
+                timestamp = ''
+                for i in range(len(row)):
+                    if yearMonthDayPattern.match(row[i]):
+                        timestamp = row[i] + ' ' + row[i+1]
+                        curRow["time"] = row[i+1]
+                        break
+
+                nesting = list()
+                prefix = ''
+                for cur in mapping:
+                    if "=" in cur:
+                        equals = cur.index('=')
+                        name = prefix + cur[0:equals]
+                        value = cur[equals+1:]
+                        if value == '{':
+                            nesting.append(prefix)
+                            prefix += name + '.'
+                        else:
+                            allStats[name] = 1
+                            castValue = -1
+
+                            #Apply  any scaling to the values
+                            matchMS = minuteSecondPattern.match(value)
+                            if matchMS:
+                                castValue = (float(matchMS.group(1)) * 60 + float(matchMS.group(2))) * 1000
+                            elif value.endswith("ns"):
+                                castValue = float(value[0:-2])/1000000
+                            elif value.endswith("us"):
+                                castValue = float(value[0:-2])/1000
+                            elif value.endswith("ms"):
+                                castValue = float(value[0:-2])
+                            elif value.endswith("s"):
+                                castValue = float(value[0:-1])*1000
```

Comment:
Another one,
TimeTotalExecute=1h04m15s
ValueError: could not convert string to float: '1h04m15'